core/src/main/java/com/yahoo/oak/InternalOakMap.java

```diff
@@ -1082,33 +973,27 @@ private OakScopedReadBuffer getNextChunkMinKey(OrderedChunk<K, V> c) {
             return upperBoundKeyForChunk;
         }
 
-        protected void advanceState() {
-
-            OrderedChunk<K, V> orderedChunk = (OrderedChunk<K, V>) getState().getChunk();
-            OrderedChunk.ChunkIter chunkIter = (OrderedChunk.ChunkIter) getState().getChunkIter();
-
-            while (!chunkIter.hasNext()) { // chunks can have only removed keys
-                orderedChunk = (OrderedChunk<K, V>) getNextChunk(orderedChunk);
-                if (orderedChunk == null) {
-                    //End of iteration
-                    setState(null);
-                    return;
-                }
-                chunkIter = getChunkIter(orderedChunk);
-            }
-
-            int nextIndex = chunkIter.next(ctx);
-            getState().set(orderedChunk, chunkIter, nextIndex);
-
-            // The boundary check is costly and need to be performed only when required,
-            // meaning not on the full scan.
-            if (chunkIter.isBoundCheckNeeded()) {
-                orderedChunk.readKeyFromEntryIndex(ctx.tempKey, nextIndex);
-                if (!inBounds(ctx.tempKey)) {
-                    setState(null);
+        @Override
+        protected boolean advanceState() {
+            boolean valueToReturn = super.advanceState();
+
+            if (valueToReturn) {
+                BasicChunk<K, V> chunk = getState().getChunk();
```

Comment:
Isn't it all part of `super.advanceState()`? Why to repeat?

Reply:
I see now that you separated the code of `advanceState()`. Please add comment explaining what is done in the base abstract class and what is done in the derived classes.

Reply:
The OakMap performs boundary check, which OakHash do not.